File: Benchmarks/Primes/IsPrime.js
function is_prime( n )
{
    if (n < 2) {
        return false;
    }

    for (var i = 2; i * i <= n; ++i) {
        if (n % i == 0) {
            return false;
        }
    }
    return true;
}

function count_primes( n )
{
    var count = 0;

    for (var i = 2; i <= n; ++i) {
        if (is_prime(i)) {
            ++count;
        }
    }
    return count;
}

limit = 100_000;
count = count_primes(limit)
console.log( "Number of primes up to " + limit + ": " + count );

File: Benchmarks/Primes/IsPrime.rb
def is_prime( n )
  return false if n < 2

  i = 2
  while i * i <= n
    return false if n % i == 0
    i += 1
  end
  true
end

def count_primes( limit )
  count = 0
  ( 2...limit ).each { |n| count += 1 if is_prime n }
  count
end

limit = 100_000
count = count_primes limit
puts "Number of primes up to #{limit}: #{count}"
